### ISSUE TITLE
Legacy method not handled correctly

### ISSUE THREAD

**Issue by ulken** [jacobbuck/react-beforeunload#27]
I haven't actually confirmed this. I've just looked at the code and tried to reason about it. I _think_ one of the legacy methods (namely `returning a string from the event handler`) is not handled properly.

Judging by the current code, it seems like this is a behavior you want to support which is why I thought I'd let you know.

Anyway, as _returning a string from the event handler_ implies, the _actual_ event handler needs to return a string. However, in this case, the [handler function](https://github.com/jacobbuck/react-beforeunload/blob/673a9f42d7e07f2764aebd36a8d2b47256b34ae0/src/useBeforeunload.js#L30) attached to the window object doesn't return anything (well `undefined`).

**Problematic code**
```js
30: const eventListener = (event) => {
31:	  eventListenerRef.current(event); // <= the return value is not forwarded
32: };
```

**Suggested fix**
```js
30: const eventListener = (event) => eventListenerRef.current(event);
```

or (a matter of preference)

```js
30: const eventListener = (event) => {
31:	  return eventListenerRef.current(event);
32: };
```

Cheers!

**Comment by jacobbuck** [comment]
Good spotting! I see it too now that you've pointed it out 😅 

Will push a fix shortly.

**Comment by ulken** [comment]
Want me to submit a PR? Do you prefer option 1 or 2?

**Comment by jacobbuck** [comment]
Sorry, I had made the change but hadn't gotten around to pushing & publishing it.

This is now fixed in [v2.5.2](https://github.com/jacobbuck/react-beforeunload/blob/master/CHANGELOG.md#v252---2021-10-03)
